### PR TITLE
service/dap: Add no-op handlers for unsupported requests

### PR DESCRIPTION
**Pull request by polinasok**
The client is notified that these requests are not supported by the false values for the corresponding capabilities in the 'initialize' response. No-op implementation is used because that is the default in vscode's codebase that defines the base classes for debug adaptors.
See https://github.com/microsoft/vscode-debugadapter-node/blob/master/adapter/src/debugSession.ts

### CODE REVIEW

File: service/dap/server_test.go [hunk 1]
```diff
@@ -216,6 +217,133 @@ func TestLaunchTestRequest(t *testing.T) {
...
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("\ngot  %#v\nwant %#v", got, want)
+		}
+		client.RestartRequest()
+		got = client.ExpectRestartResponse(t)
+		want = &dap.RestartResponse{Response: newResponse(3, "restart")}
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("\ngot  %#v\nwant %#v", got, want)
+		}
+		client.SetFunctionBreakpointsRequest()
```
Comment (eliben) [inline service/dap/server_test.go:400]: Nit: add an empty line between each request test (after the closing `}` of the comparison `if`, to make them a little easier to distinguish visually

**Review by eliben — APPROVED** [review]

File: service/dap/server_test.go (hunk 1 above)
Comment (polinasok) [reply service/dap/server_test.go:400]: Done

**Comment by polinasok** [comment]
Now tracked as https://github.com/go-delve/delve/pull/1918

**Comment by eliben** [comment]
Consider closing PRs here once they're tracked upstream. Any reason not to do that?

**Pull request closed without merging**